### PR TITLE
fix(fs): revert dummy column names to normal file path before export

### PR DESCRIPTION
**Pull request by shinyano**
# what's wrong:
Filenames in dummy dir will be renamed and stored in fs. 
'xxx/dir/file.txt' will be stored as 'dir.file\txt' in IGinX to comply with column name rules.

When excuting "select * from dir into outfile "test" as stream", client attempts to create 'test\dir.file\txt' because the query result is given directly. As dir 'dir.file' does not exist, exception will be raised. Also the dir structure in 'test\dir.file\txt' is wrong.

# fix:
Revert filenames to original file path before exporting.
In same example above, client will attempt to create 'test\dir\file.txt'.

# other engine:
export test(s1,s2) creates two file: s1 & s2 in test dir.
e.g. "select * from test into outfile "test" as stream" creats 'test\test\s1' and 'test\test\s2'

### CODE REVIEW

File: client/src/main/java/cn/edu/tsinghua/iginx/client/IginxClient.java [hunk 1]
```diff
@@ -509,6 +510,8 @@ private static void processExportByteStream(QueryDataSet res)
         finalCnt--;
         continue;
       }
+      originColumn = swapChar(originColumn, '\\', '.');
+      originColumn = originColumn.replace('\\', File.separatorChar);
```
Comment (zhuyuqing) [inline client/src/main/java/cn/edu/tsinghua/iginx/client/IginxClient.java:514]: 这个好像还是有点问题，我们允许有schemaPrefix，所以，这么搞会把SchemaPrefix也变成目录的一级，这样符合用户的期望吗？

如果是导出一般的序列，是不是也会变成多级目录？如果用户希望保留序列名的话，那不就很麻烦了？

**Review by zhuyuqing — CHANGES_REQUESTED** [review]

File: shared/src/main/java/cn/edu/tsinghua/iginx/utils/FileUtils.java [hunk 2]
```diff
@@ -41,4 +48,34 @@ public static void exportByteStream(List<List<byte[]>> values, String[] columns)
...
+        sb.append(a);
+      } else {
+        sb.append(str.charAt(i));
+      }
+    }
+    return sb.toString();
+  }
+
+  // create dir recursively
+  private static boolean createDirectory(String directoryPath) {
```
Comment (SolomonAnn) [inline shared/src/main/java/cn/edu/tsinghua/iginx/utils/FileUtils.java:68]: mkdirs()函数能直接递归创建目录

File: client/src/main/java/cn/edu/tsinghua/iginx/client/IginxClient.java (hunk 1 above)
Comment (shinyano) [reply client/src/main/java/cn/edu/tsinghua/iginx/client/IginxClient.java]: 改为保留序列名，不考虑目录层次

File: shared/src/main/java/cn/edu/tsinghua/iginx/utils/FileUtils.java (hunk 2 above)
Comment (shinyano) [reply shared/src/main/java/cn/edu/tsinghua/iginx/utils/FileUtils.java]: 已修改

**Review by SolomonAnn — APPROVED** [review]

**Pull request merged**